source/common/router/router.cc

```diff
@@ -487,7 +488,25 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::HeaderMap& headers, bool e
   }
 
   // Fetch a connection pool for the upstream cluster.
-  Http::ConnectionPool::Instance* conn_pool = getConnPool();
+  Http::ConnectionPool::Instance* conn_pool;
+  const auto http_protocol_options = cluster_->httpProtocolOptions();
+
+  if (http_protocol_options != absl::nullopt && http_protocol_options.value().auto_sni()) {
+    auto url_str = headers.Host()->value().getStringView();
+    const auto parsed_authority = Http::Utility::parseAuthority(url_str.data());
```

Comment:
url_str is absl::string_view type.
The parameter should be `url_str`

Reply:
+1